Sources/TSCBasic/Lock.swift

```diff
@@ -11,6 +11,11 @@
 import Foundation
 import TSCLibc
 
+public enum LockType {
```

Comment:
should this renamed to `FileLockType` or nested under `FileLock`?

Reply:
Let's put it inside of `FileLock`

Reply:
I see your point, but personally prefer to follow the rule of three with generalizations. since this is not a stable API I am not too worried about API surface area, but as a rule of thumb introducing top level types is something to do with care